jina/drivers/multimodal.py

```diff
@@ -0,0 +1,114 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from collections import defaultdict
+from typing import Iterable, Tuple, Dict
+
+import numpy as np
+
+from .encode import BaseEncodeDriver
+from .helper import pb2array, array2pb
+from ..proto import jina_pb2
+
+
+def _extract_doc_content(doc: 'jina_pb2.Document'):
+    """Returns the content of the document with the following priority:
+    If the document has an embedding, return it, otherwise return its content.
+    """
+    if doc.embedding.buffer:
+        return pb2array(doc.embedding)
+    else:
+        return doc.text or doc.buffer or (doc.blob and pb2array(doc.blob))
```

Comment:
is this somehow related to https://github.com/jina-ai/jina/blob/521436c17a3a807bb47c020a4527b209651624c4/jina/drivers/helper.py#L90-L93

if yes, can we refactor it to unify them into one?

Reply:
it is kind of the same, it differs I think in the way documents are passed. Since here we are looping on chunks internall. I thought about it but did not want to overcomplicate.